go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -998,19 +998,22 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 						return fmt.Errorf("unable to get link for %s, error: %v", types.K8sMgmtIntfName, err)
 					}
 					var gwIP net.IP
+					var srcIP net.IP
 					var routes []routemanager.Route
 					for _, subnet := range config.Kubernetes.ServiceCIDRs {
 						if utilnet.IsIPv4CIDR(subnet) {
 							gwIP = mgmtPortConfig.ipv4.gwIP
+							srcIP = config.Gateway.MasqueradeIPs.V4HostMasqueradeIP
 						} else {
 							gwIP = mgmtPortConfig.ipv6.gwIP
+							srcIP = config.Gateway.MasqueradeIPs.V6HostMasqueradeIP
 						}
 						subnet := *subnet
 						routes = append(routes, routemanager.Route{
 							GwIP:   gwIP,
 							Subnet: &subnet,
 							MTU:    config.Default.RoutableMTU,
-							SrcIP:  nil,
+							SrcIP:  srcIP,
```

Comment:
Unsure if i should use the host masq IP here when the interface is the mgnt port.

Reply:
It doesn't make sense. Removed.